src/components/filterBox/tests/FilterBox.spec.tsx

```diff
@@ -47,6 +47,18 @@ describe('FilterBox', () => {
             expect(renderSpy.calls.count()).toBe(1);
         });
 
+        it('should call prop onRender with the id and the matchFilter to set in the store', () => {
+            const renderSpy = jasmine.createSpy('onRender');
+            const matchFilter = jasmine.createSpy('matchFilter');
+
+            expect(() => filterBoxInstance.componentWillMount()).not.toThrow();
+
+            filterBox.setProps({id: id, onRender: renderSpy, matchFilter: matchFilter});
+            filterBox.unmount();
+            filterBox.mount();

```

Comment:
You really need to unmount and mount after a setProps? If it doesn't work, maybe try an update after instead